server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1220,13 +1219,7 @@ public void run() {
             }
             // Reschedule itself to run again if not closed
             if (closed.get() == false) {
```

Comment:
is the `closed` AtomicBoolean still needed?

Reply:
I wasn't (and still am not) sure about that: Then this thing would only stop scheduling if you shut down the thread pool which might lead to weird/unexpected behavior in tests? Seems nicer to reason about to me to not make it rerun when the indices service was  closed/stopped.